### ISSUE TITLE
Add Support for Custom Submit Button Text

### ISSUE THREAD

**Issue by sarahwcodes** [department-of-veterans-affairs/component-library#1711]
Problem: The `va-search-input` component currently does not allow developers to customize the `aria-label` of the submit button. This creates accessibility issues, as the button lacks discernible text for screen readers and registers as a critical error in Axe DevTools (see screenshot below; library is being used for VA Notify platform).

Potential Solutions:
- [ ] Add a `submit-button-aria-label` (or similar) property to allow developers to set an accessible label for the button.

<img width="1400" height="736" alt="Image" src="https://github.com/user-attachments/assets/4cd037ef-e5d7-4f41-8616-28c07b19e7d3" />
